src/map/zone.cpp

```diff
@@ -1174,6 +1174,15 @@ void CZone::CharZoneOut(CCharEntity* PChar)
         PChar->PTreasurePool->DelMember(PChar);
     }
 
+    // If zone-wide treasure pool but no players in zone then destroy current pool and create new pool
+    // this prevents loot from staying in zone pool after the last player leaves the zone
+    if (m_TreasurePool && m_TreasurePool->GetPoolType() == TREASUREPOOL_ZONE && m_zoneEntities->CharListEmpty())
+    {
+        destroy(m_TreasurePool);
+        delete m_TreasurePool;
```

Comment:
This is now a double delete, which is UB

```cpp
template <typename T>
inline void destroy(T*& ptr)
{
    delete ptr; // cpp.sh allow
    ptr = nullptr;
}
```

Reply:
> This is now a double delete, which is UB
> 
> ```c++
> template <typename T>
> inline void destroy(T*& ptr)
> {
>     delete ptr; // cpp.sh allow
>     ptr = nullptr;
> }
> ```

Ah, true, thanks, I have updated the PR.